src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -1605,6 +1641,9 @@ static void uct_ib_mlx5_devx_check_odp(uct_ib_mlx5_md_t *md,
     ucs_debug("%s: ODP is supported, version %d",
               uct_ib_device_name(&md->super.dev), version);
     md->super.reg_nonblock_mem_types = UCS_BIT(UCS_MEMORY_TYPE_HOST);
+    if ((version == 2) || (md_config->devx_objs == 0)) {
```

Comment:
why need to check for `md_config->devx_objs == 0`?
isn't this code skipped if both versions of odp are not supported?

Reply:
odpv2 should support devx
when devx is disabled odpv1 is enough

Reply:
yes, but it is already checked above in this func. If odp is v1 and devx objects var contains rcqp there is a goto which skips this code. Imo, we do not need to check devx objects here, because we already checked above that one of ODP versions is supported